pgjdbc/src/main/java/org/postgresql/jdbc/PgPreparedStatement.java

```diff
@@ -626,9 +627,15 @@ public void setObject(int parameterIndex, Object in, int targetSqlType, int scal
         setString(parameterIndex, castToString(in), Oid.BPCHAR);
         break;
       case Types.VARCHAR:
-      case Types.LONGVARCHAR:
         setString(parameterIndex, castToString(in), getStringType());
         break;
+      case Types.LONGVARCHAR:
+        if (in instanceof InputStream) {
+          preparedParameters.setText(parameterIndex, (InputStream)in);
```

Comment:
Test database is created here: https://github.com/pgjdbc/pgjdbc/blob/master/.travis.yml#L10

We can even use different encodings if that is required for tests.

If a particular test requires the db to be UTF-8, it should be handled as `Assume.assumeTrue(con....)` kind of validation in the test setup like this: https://github.com/pgjdbc/pgjdbc/blob/05c2f8d0d5ad66d9bf659754483ef92acb1b30ff/pgjdbc/src/test/java/org/postgresql/test/jdbc4/JsonbTest.java#L29

When the "assumption" fails, the test is just ignored. The test should be a junit 4 test (e.g. `extends BaseTest4`)